pkg/collector/worker/worker.go

```diff
@@ -148,6 +155,25 @@ func (w *Worker) Run() {
 
 		utilizationTracker.CheckStarted()
 
+		if checkStats, found := expvars.CheckStats(check.ID()); found {
+
+			// Calculate start time of previous execution
+			previousFinishDate := checkStats.LastSuccessDate
+			if previousFinishDate < checkStats.LastErrorDate {
+				previousFinishDate = checkStats.LastErrorDate
+			}
+
+			previousStartDate := previousFinishDate - (checkStats.LastExecutionTime / 1e3)
+
+			delay := time.Now().Unix() - previousStartDate - int64(check.Interval().Seconds())
+
+			if delay < 0 {
+				delay = 0
+			}
+
+			checkDelay.Set(float64(delay), check.String())
+		}
```

Comment:
Can you move this whole block to `check/stats/stats.go` please? I think it can go in `Add()`, but if it absolutely must be done at check start, we can add another method.